File: ruby/interviewbit/anagram.rb
#https://www.geeksforgeeks.org/check-whether-two-strings-are-anagram-of-each-other/



a = ["cat", "dog", "tac", "god", "act"]

words_hash = Hash.new{|h,k| h[k] = []}

for val in a do
  sort_word = val.chars.sort.join

  words_hash[sort_word].push(val)
end

p words_hash
  # words_hash.each {|key, val| p val}

File: python/interviewbit/anagram.py
from collections import defaultdict

def are_anagrams(a):
  anagrams_dict = {}
  
  
  for x in a:
    temp = x
    sort_str = ''.join(sorted(x))
    
    value = anagrams_dict.get(sort_str)
    
    if not value:
      anagrams_dict[sort_str] = []
    anagrams_dict[sort_str].append(temp)
  
    # print(anagrams_dict)
  print(anagrams_dict)
    
are_anagrams(a = ["cat", "dog", "tac", "god", "act"])